wicker/core/datasets.py

```diff
@@ -152,33 +225,42 @@ def get_folder_size(bucket, prefix):
         return get_folder_size(bucket, key)
 
     def _get_dataset_size(self):
-        """Gets total size of the dataset in bits
+        """Gets total size of the dataset in bits.
 
         Returns:
             int: total dataset size in bits
         """
+
         # intialize with size of parquet dir
+        print("Parsing parquet and arrow dir for size.")
         par_dir_bytes = self._get_parquet_dir_size()
 
         # need to know which columns are heavy pntr columns we go to for
-        # byte adding
+        # byte adding, parse which are heavy pointers based off metadata
+        # baked into schema
         schema = self.schema()
         heavy_pointer_cols = []
         for col_name in schema.get_all_column_names():
-            if schema.get_column(col_name).is_heavy_pointer:
-                heavy_pointer_cols.append(col_name)
+            column = schema.get_column(col_name)
+            if column is not None:
+                if column.is_heavy_pointer:
+                    heavy_pointer_cols.append(col_name)
 
-        # Each individual row only knows which column file it goes to, so we have to
-        # neccesarily parse all rows :( to get the column files. This should be cached
-        # as metadata but that would require re-curating the datasets.
+        # create arrow table for parsing
         print("Creating arrow table")
         arrow_table = self.arrow_table()
 
         buckets_keys = set()
-        worker = ShuffleWorker(storage=self._storage)
+
+        # ignore typing to avoid changing the typing of Shuffle Worker yet
+        # ToDo: Change typing of ShuffleWorker and let it take local data storage
+        worker = ShuffleWorker(storage=self._storage)  # type: ignore
         print("Processing through heavy pointers")
         for heavy_pntr_col in heavy_pointer_cols:
             print(f"Evaulating {heavy_pntr_col} for column file locations")
+            # have to go through all data in arrow table col
```

Comment:
@pickles-bread-and-butter could you actually take Jay's original comment from the old line 171 and just stuff it here (replacing your comment)? I like how he explained it (although of course I don't like the fact that we have to go through all of the rows LOL).